### PR TITLE
Improve document v1 visiting selection handling

### PR DESCRIPTION
**Pull request by vekterli**
@freva please review. This addresses some long-standing shortcomings of the V1 API in terms of selection handling.

### CODE REVIEW

File: vespaclient-container-plugin/src/test/java/com/yahoo/document/restapi/resource/RestApiTest.java [hunk 1]
```diff
@@ -277,29 +280,105 @@ public void testbasicVisit() throws Exception {
...
-    String visit_test_bad_response = "Visiting does not support setting value for group/value in combination with expression";
-
+    String visit_test_uri_selection_rewrite = "/document/v1/namespace/document-type/group/abc?continuation=abc";
+    String visit_test_response_selection_rewrite = "doc selection: 'id.group=='abc''";
 
     @Test
-    public void testBadVisit() throws Exception {
-        Request request = new Request("http://localhost:" + getFirstListenPort() + visit_test_bad_uri);
+    public void testUseExpressionOnVisit() throws Exception {
+        Request request = new Request("http://localhost:" + getFirstListenPort() + visit_test_uri_selection_rewrite);
```
Comment (freva) [inline vespaclient-container-plugin/src/test/java/com/yahoo/document/restapi/resource/RestApiTest.java]: This test can be simplified to
```
assertResultingDocumentSelection("group/abc?continuation=abc", "id.group=='abc'");
```
?

File: vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/RestApi.java [hunk 2]
```diff
@@ -278,27 +281,81 @@ String getParameter() {
...
             }
         }
+        return documentSelection;
+    }
+
+    private HttpResponse handleVisit(RestUri restUri, HttpRequest request) throws RestApiException {
+        String documentSelection;
         OperationHandler.VisitOptions options;
         try {
+            documentSelection = documentSelectionFromRequest(restUri, request);
```
Comment (freva) [inline vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/RestApi.java]: Why was there 1 `=`  in `documentSelection` before, but 2 `=` now?

Comment (freva) [reply vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/RestApi.java]: The value of the (old) `documentSelection` previously had one equals sign between, for example, `id.group` and its value (https://github.com/vespa-engine/vespa/blob/177392cd29ab9db2de0471d295cce0d0423ed726/vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/RestApi.java#L297)

But now there are two equal signs (https://github.com/vespa-engine/vespa/blob/932812ddcd333ce57c0a5d812b12d4f0ced971a5/vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/RestApi.java#L324)

Comment (vekterli) [reply vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/RestApi.java]: Ahh yes, sorry 🙂 Basically, the `=` operator in document selection is a multi-purpose operator which for integers degenerates to `==`, but for strings can end up as glob-style operator that uses regexes behind the scenes. Operator `==` always implies a concrete match, which is the semantics we want here.

I changed this to ensure we have a deterministic performance profile for comparisons. No reason to use `=` here in the first place, though it _does_ functionally work as intended.

**Review by freva — APPROVED** [review]

**Pull request merged**